presto-main/src/main/java/com/facebook/presto/util/Types.java

```diff
@@ -32,4 +32,15 @@ private Types() {}
                 value.getClass().getName());
         return target.cast(value);
     }
+
+    /**
+     * Uses 64 bit hash and compact its entropy in 32 bits.
+     * @deprecated This function should only be used for keeping backward compatibility with external libraries using 32 hashes.
```

Comment:
Ah, ok. The comment was not very clear. Maybe add that explanation to the javadoc.